src/data-loader.ts

```
@@ -1347,62 +1496,82 @@ export async function loadSessionBlockData(
 	const allEntries: LoadedUsageEntry[] = [];
 
-	for (const file of sortedFiles) {
-		const content = await readFile(file, 'utf-8');
-		const lines = content
-			.trim()
-			.split('\n')
-			.filter(line => line.length > 0);
+	const filePromises = sortedFiles.map(async (file) => {
+		const fileEntries = await fileCache.getOrProcess(file, async (file, _baseDir) => {
+			const entries: LoadedUsageEntry[] = [];
+			const content = await readFile(file, 'utf-8');
```

ESLint

[error] Unsafe assignment of an error typed value.

(ts/no-unsafe-assignment)